### PR TITLE
feat(axios): use named parameters

### PR DESCRIPTION
**Pull request by Marcel-G**
## Status

**READY**

## Description

Often times routes have many path parameters. As the number grows, type-safety is less effective as arguments (commonly string or number types) can easily get used in the wrong order.
Using named parameters helps this, as you get the added safety of needing to specify the appropriate name for the parameter.

This PR adds a `useNamedParameters` option for the axios client which creates path parameters as a single object.

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.

```bash
> git pull --prune
> git checkout Marcel-G:feat/named-parameters
> yarn build
> cd /tests
> yarn generate
> vi generated/axios/named-parameters/endpoints.ts
```

Heres an example output:

```ts
/**
 * @summary Info for a specific pet
 */
export const showPetById = <TData = AxiosResponse<Pet>>(
  { petId, version = 1 }: ShowPetByIdPathParameters, options?: AxiosRequestConfig
): Promise<TData> => {
  return axios.get(
    `/v${version}/pets/${petId}`, options
  );
}
type ShowPetByIdPathParameters = {
  petId: string,
  version?: number,
}
```

### CODE REVIEW

File: docs/src/pages/reference/configuration/output.md [hunk 1]
```diff
@@ -671,6 +671,36 @@ Default Value: `'root'`.
 
 Can be used to set the value of `providedIn` on the generated Angular services. If `false`, no `providedIn` will be set. If `true` or not specified, it will fall back to the default value: `root`.
 
+#### axios
+
+Type: `Object`.
+
+Give you specific options for the axios client
```
Comment (Marcel-G) [inline docs/src/pages/reference/configuration/output.md]:
```suggestion
Give specific options for the axios client
```

File: docs/src/pages/reference/configuration/output.md [hunk 2]
```diff
@@ -671,6 +671,36 @@ Default Value: `'root'`.
...
+
+```js
+module.exports = {
+  petstore: {
+    output: {
+      ...
+      override: {
+        angular: {
```
Comment (Marcel-G) [inline docs/src/pages/reference/configuration/output.md]:
```suggestion
        axios: {
```

**Comment by anymaniax** [comment]
it's doesn't seems to work properly when there is no path params

**Comment by Marcel-G** [comment]
@anymaniax this change seems to have some overlap with https://github.com/anymaniax/orval/pull/396. Do you have a preference between the two implementations?

**Comment by anymaniax** [comment]
Hello @Marcel-G and thanks a lot. Seems great! Would you be interested to add a global config for that and do it for all client instead?

**Comment by anymaniax** [comment]
@Marcel-G do you think we can merge the other one directly?

**Comment by Marcel-G** [comment]
> @Marcel-G do you think we can merge the other one directly?

Actually this PR goes with a slightly different approach so the other PR is redundant if we go this way.

**Comment by anymaniax** [comment]
@Marcel-G for me other one make more sense to have a more global way of doing it. What do you think?

**Comment by anymaniax** [comment]
@Marcel-G can you also contact me on discord when you have the time?

**Pull request closed without merging**